ui/src/source.ts

```diff
@@ -224,3 +221,28 @@ export const findReadme = (tree: Tree): string | null => {
   }
   return null;
 }
+
+export const readme = (
+  projectId: string,
+  revision: string,
+): Readable<remote.Data<Readme | null>> => {
+  const readme = remote.createStore<Readme | null>();
+
+  remote
```

Comment:
This bit is a little 🤯 for me; can you provide a quick tldr of what the chain is doing here?

Reply:
`chain` ties one store to another, advancing their loading state in lockstep, and setting the error status if there was one in the dependent store. Once the dependent store has transitioned to `Success` the promise is resolved, that we use here to see  if the loaded object is a tree, search for a readme and load that readme.